frontend/src/static/js/v_zoom.js

```diff
@@ -10,6 +14,22 @@ window.vZoom = {
     getSliceLink(slice) {
       return `${window.getBaseLink(this.info.user.repoId)}/commit/${slice.hash}`;
     },
+
+    updateUser(repos) {
+      const userOrig = repos[this.info.repoIndex][this.info.userIndex];
+      const { commits } = userOrig;
+      const commitsWithinDuration = commits.filter(
+          (commit) => commit.date >= this.info.sinceDate && commit.date <= this.info.untilDate,
```

Comment:
The date of commits has already been filtered, see [here](https://github.com/reposense/RepoSense/blob/master/frontend/src/static/js/v_summary.js#L562). So maybe one is enough.

Reply:
The openTabZoom method is not called when user switches between the granularity, it only updates the `info.user` and `filterTimeFrame` within the zoom component. 

Since I am passing the list of commits that has not been filtered yet to the zoom component, I will need to filter commits that within the selected zoom range.

Reply:
If it is the case, we should only process the commits in `v_zoom`, even when using `openTabZoom`. You should modify the existing logic to make the whole logic more fluent.